reference/5.1/Microsoft.PowerShell.Core/About/about_PowerShell_exe.md

```diff
@@ -47,9 +47,9 @@ PowerShell[.exe] -Help | -? | /?
 
 ### -Command
 
-Executes the specified commands (and any parameters) as though they were typed
-at the PowerShell command prompt, and then exits, unless the `NoExit`
-parameter is specified.
+Executes the specified commands (and any parameters), one statement at a time,
+as though they were typed at the PowerShell command prompt, and then exits,
+unless the `NoExit` parameter is specified.
```

Comment:
I appreciate the update, but please note that it is only `-` as the argument, i.e. _stdin_ input that exhibits the pseudo-interactive, each-newline-separated-statement-on-its-own behavior.

In other words: passing a command string to `-Command` or a file path to `File` does _not_ exhibit this behavior, and code execution works as expected (execution as a _single script block_).

Reply:
Consider the difference between the following two commands:

```powershell
# Sample code that normally makes the 1 / 0 error abort the whole script block.
$commandText = @'
$ErrorActionPreference = 'Stop'
1 / 0
'done'
'@ 

# OK - execution as single script block; 'done' never prints.
pwsh -nop -nolog -Command $commandText

'---'


# !! Pseudo-interactive behavior.
# 'done' still prints, because `$ErrorActionPreference = 'Stop'` was executed separately
# and therefore had no effect on `1 / 0`
$commandText | pwsh -nop -nolog -Command -
```

Note while the `-Command -` is _somewhat_ like one-by-one, interactive submission, in the latter case `$ErrorActionPreference = 'Stop'` _would_ be remembered for subsequent submissions.